mlflow/llama_index/__init__.py

```diff
@@ -0,0 +1,385 @@
+        mlflow_model: An MLflow model object that specifies the flavor that this model is being
+            added to.
+        signature: A Model Signature object that describes the input and output Schema of the
+            model. The model signature can be inferred using `infer_signature` function
+            of `mlflow.models.signature`.
+        input_example: {{ input_example }}
+        pip_requirements: {{ pip_requirements }}
+        extra_pip_requirements: {{ extra_pip_requirements }}
+        conda_env: {{ conda_env }}
+        metadata: {{ metadata }}
+        : Optional additional configurations for transformers serialization.
+    """
+    import llama_index
+
+    if engine_type not in _SUPPORTED_ENGINES:
+        raise ValueError(
+            f"Currently mlflow only supports the following engine types: "
+            f"{_SUPPORTED_ENGINES}. {engine_type} is not supported, so please"
+            "use one of the above types"
+        )
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    path = os.path.abspath(path)
+    _validate_and_prepare_target_save_path(path)
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, path)
+
+    if signature is None and input_example is not None:
+        wrapped_model = _LlamaIndexModelWrapper(index)
+        signature = infer_signature_from_input_example(input_example, wrapped_model)
+    elif signature is False:
+        signature = None
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        signature = validate_and_resolve_signature(index, signature)
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    engine_kwargs_path = os.path.join(path, _ENGINE_KWARGS_DIRECTORY)
+    serialize_engine_kwargs_to_json(engine_kwargs, engine_kwargs_path)
+
+    from llama_index.core import Settings
+
+    settings_path = os.path.join(path, _SETTINGS_DIRECTORY)
+    serialize_settings_to_json(Settings, settings_path)
+
+    model_data_path = path
+    _save_model(index, model_data_path)
+
+    flavor_conf = {"engine_type": engine_type}
+    if _should_add_pyfunc_to_model(index):
+        if mlflow_model.signature is None:
+            mlflow_model.signature = infer_signature_from_input_example(
+                index=index,
+                example=input_example,
+                flavor_config=flavor_conf,
+            )
+
+    pyfunc.add_to_model(
+        mlflow_model,
+        loader_module="mlflow.llama_index",
+        conda_env=_CONDA_ENV_FILE_NAME,
+        python_env=_PYTHON_ENV_FILE_NAME,
+        code=code_dir_subpath,
+    )
+    mlflow_model.add_flavor(
+        FLAVOR_NAME,
+        llama_index_version=llama_index.core.__version__,
+        code=code_dir_subpath,
+        **flavor_conf,
+    )
+    if size := get_total_file_size(path):
```

Comment:
Validate with the new Resources object type to determine if this is the correct way of doing this now. Parameters that are sent to model serving for validating instance type for serving considerations should conform to the standard interface.